### PR TITLE
 refactor: remove unused storage traces 

### PR DESCRIPTION
**Pull request by omerfirmak**
## 1. Purpose or design rationale of this PR

reduces the tracing overhead from %16 -> %14-11, so overall its barely noticable. But I hope to make a series of small improvements which then combined will result in a noticable speed up.

triggered a good amount of code to be deleted as well.

## 2. PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [ ] build: Changes that affect the build system or external dependencies (example scopes: yarn, eslint, typescript)
- [ ] ci: Changes to our CI configuration files and scripts (example scopes: vercel, github, cypress)
- [ ] docs: Documentation-only changes
- [ ] feat: A new feature
- [ ] fix: A bug fix
- [ ] perf: A code change that improves performance
- [ ] refactor: A code change that doesn't fix a bug, or add a feature, or improves performance
- [ ] style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
- [ ] test: Adding missing tests or correcting existing tests

## 3. Deployment tag versioning

Has the version in `params/version.go` been updated?

- [ ] This PR doesn't involve a new deployment, git tag, docker image tag, and it doesn't affect traces
- [ ] Yes

## 4. Breaking change label

Does this PR have the `breaking-change` label?

- [ ] This PR is not a breaking change
- [ ] Yes

### CODE REVIEW

**Comment by lispc** [comment]
another tip is: for our usage (proving and ccc), we always set "l.cfg.DisableStorage" to true.   We don't need storage.   Keeping this config is mainly for compatibility to standard eth RPC API.

File: eth/tracers/native/prestate.go [hunk 1]
```diff
@@ -119,9 +121,17 @@ func (t *prestateTracer) CaptureState(pc uint64, op vm.OpCode, gas, cost uint64,
...
 	caller := scope.Contract.Address()
 	switch {
-	case stackLen >= 1 && (op == vm.SLOAD || op == vm.SSTORE):
+	case stackLen >= 1 && op == vm.SSTORE:
 		slot := common.Hash(stackData[stackLen-1].Bytes32())
 		t.lookupStorage(caller, slot)
+	case stackLen >= 1 && op == vm.SLOAD:
+		slot := common.Hash(stackData[stackLen-1].Bytes32())
+		t.inflightSload = func(readValue common.Hash) {
+			if _, ok := t.pre[caller].Storage[slot]; ok {
```
Comment (lispc) [inline eth/tracers/native/prestate.go]: <del>when will this "ok" be true? multi sload in same tx/block? </del> i think so

File: core/vm/logger.go [hunk 2]
```diff
@@ -304,6 +302,11 @@ func (l *StructLogger) CaptureState(pc uint64, op OpCode, gas, cost uint64, scop
 }
 
 func (l *StructLogger) CaptureStateAfter(pc uint64, op OpCode, gas, cost uint64, scope *ScopeContext, rData []byte, depth int, err error) {
+	stack := scope.Stack
+	if op == SLOAD && stack.len() >= 1 {
+		storageValue := stack.data[stack.len()-1].Bytes32()
+		l.inflightSload(storageValue)
```
Comment (lispc) [inline core/vm/logger.go]: i noticed you set `inflightSload` to nil after being used in pre state tracer.  Why not here?

**Comment by lispc** [comment]
1. we could also separate this into 2 prs. remove the zktrie witness deadcode, and the inflightSload optimization. They are not too closely related. Anyway, this would also be ok.
2. How will you plan to test this change?   ensure traces before/after this pr are same for a block?

**Review by lispc — DISMISSED** [review]

File: core/vm/logger.go (hunk 2 above)
Comment (0xmountaintop) [reply core/vm/logger.go]: I think we'd better also need to set it to nil here, I guess Omer forgets

**Comment by 0xmountaintop** [comment]
I think you can mention in the PR description (https://github.com/scroll-tech/go-ethereum/pull/819#issue-2346601401) that we remove storage tracing here: storage tracing is only needed long time ago, we don't need it any more (cannot remember since which version but at least https://github.com/scroll-tech/go-ethereum/pull/321)

**Review by 0xmountaintop — DISMISSED** [review]

**Comment by omerfirmak** [comment]
> we could also separate this into 2 prs. remove the zktrie witness deadcode, and the inflightSload optimization. They are not too closely related. Anyway, this would also be ok.

yeah, I will leave just deadcode clean up in this PR and worry about performance related stuff after Curie.

**Review by 0xmountaintop — APPROVED** [review]

**Review by georgehao — APPROVED** [review]

**Pull request merged**